cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -734,6 +735,18 @@ pub(crate) fn define(
     }
     e.enc64(bconst.bind(B64), rec_pu_id_bool.opcodes(vec![0xb8]).rex());
 
+    let is_zero_int = InstructionPredicate::new_is_zero_int(f_unary_imm, "imm");
+    e.enc_both_instp(
+        iconst.bind(I32),
```

Comment:
For what it's worth, if there aren't any encodings for these types, legalization will properly create new i32 constants and reduce them.

In the i8 test that's failing, I suspect this changes the number of live values, so adapting to the new numbers should work. Ideally, we'd use the regexp directive and would catch the value number with the regexp, as is done in a few other tests now.

For i16, we could use `rec_u_id_z.opcodes.(vec![0x66, 0x31])` instead, but as @bjorn3 said, upper halves are undefined in this case, and since the instruction operating on larger types is smaller than the one operating on i16, this seems better to use the xor $eax encoding. How about updating the test, and adding a comment (here, and maybe in the test) that the i32 encoding is smaller?

Reply:
Okay, done and done!